### PR TITLE
Support thor 1.Y+

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: librarianp.gemspec [hunk 1]
```diff
@@ -17,9 +17,9 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
 
-  gem.add_dependency "thor", "~> 0.15"
+  gem.add_dependency "thor", "~> 1.0"
 
-  gem.add_development_dependency "rake"
+  gem.add_development_dependency 'rake', '< 11.0'
```
Comment (bastelfreak) [inline librarianp.gemspec:22]: whats the reason for pining to such an old version?

Comment (ehelms) [reply librarianp.gemspec:22]: Effectively this https://stackoverflow.com/questions/35893584/nomethoderror-undefined-method-last-comment-after-upgrading-to-rake-11

If I upgrade rspec, I hit a variety of test errors. I can address that here and increase the overall scope of these changes.

This change is designed to allow us to be able to use Facter 4 and librarian-puppet in the same project. Facter 4 relies on Thor 1+.

**Comment by ehelms** [comment]
Older rubies and newer rubies resolving the sorting differently is bothersome for that one single test. Do we actually care that they are in a particular order vs being present?

File: Gemfile [hunk 2]
```diff
@@ -3,6 +3,6 @@ source "https://rubygems.org"
 # Specify your gem's dependencies in librarian.gemspec
 gemspec
 
-gem "fakefs"
+gem "fakefs", '~> 0.4.2'
```
Comment (ekohl) [inline Gemfile]: This is already a development dependency. Isn't this redundant?

**Comment by ekohl** [comment]
I started with https://github.com/voxpupuli/librarian/pull/11 and I saw the same failure you see. This confirms that it's best to swap them I think.

File: librarianp.gemspec (hunk 1 above)
Comment (ghoneycutt) [reply librarianp.gemspec:22]: We should not introduce change that pins a dependency to old an version with known CVE's.

https://github.com/advisories/GHSA-jppv-gw3r-w3q8

**Comment by ekohl** [comment]
Closing in favor of https://github.com/voxpupuli/librarian/pull/11

**Pull request closed without merging**